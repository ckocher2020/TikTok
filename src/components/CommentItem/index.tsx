import React from 'react';
import { View, StyleSheet, Text, Image } from 'react-native';
import { CommentType } from '../../utils/types';

const CommentItem = ({ comment }: { comment: CommentType }) => (
  <View style={styles.container}>
    <View style={styles.avatarContainer}>
      <Image
        style={styles.avatarImage}
        source={require('../../../assets/avatar.png')}
      />
    </View>
    <View style={styles.commentContainer}>
      <Text style={styles.commentText}>{comment.content}</Text>
      <Text style={styles.dateText}>{comment.date}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 10,
    flexDirection: 'row',
    marginBottom: 6
  },
  avatarContainer: {
    width: 40,
    height: 40,
    alignItems: 'center',
    justifyContent: 'center'
  },
  avatarImage: {
    width: 40,
    height: 40,
    borderRadius: 20
  },
  commentContainer: {
    flex: 1,
    paddingLeft: 5,
    flexDirection: 'row',
    alignItems: 'baseline'
  },
  commentText: {
    flex: 1,
    fontSize: 18,
    color: 'black'
  },
  dateText: {
    marginLeft: 10,
    color: 'rgba(0, 0, 0, 0.5)',
    alignSelf: 'flex-end'
  }
});

export default CommentItem;
