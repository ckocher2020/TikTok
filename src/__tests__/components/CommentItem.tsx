import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CommentItem from '../../components/CommentItem';

it('CommentItem renders correctly with defaults', () => {
  const comment = {
    sender: 'James',
    date: '6: 30 am',
    content: 'Comment Item Jest'
  };
  const commentItem = renderer
    .create(<CommentItem comment={comment} />)
    .toJSON();

  expect(commentItem).toMatchSnapshot();
});
