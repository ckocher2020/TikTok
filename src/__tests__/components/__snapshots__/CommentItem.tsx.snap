// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 6,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/avatar.png",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "baseline",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "flex": 1,
          "fontSize": 18,
        }
      }
    >
      Comment Item Jest
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "rgba(0, 0, 0, 0.5)",
          "marginLeft": 10,
        }
      }
    >
      6: 30 am
    </Text>
  </View>
</View>
`;

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 6,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/avatar.png",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "baseline",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "flex": 1,
          "fontSize": 18,
        }
      }
    >
      Comment Item Jest
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "rgba(0, 0, 0, 0.5)",
          "marginLeft": 10,
        }
      }
    >
      6: 30 am
    </Text>
  </View>
</View>
`;
